src/primitives/transaction.cpp

```diff
@@ -1050,6 +1050,14 @@ bool CTransaction::ContextualCheck(CValidationState& state, int nHeight, int dos
     if (!CheckBlockAtHeight(state, nHeight, dosLevel))
         return false;
 
+    // after this fork point we do not have any transaction flowing but coinbase txes
+    if (!IsCoinBase() && ForkManager::getInstance().areTransactionsStopped(nHeight)) {
+        LogPrintf("%s():%d - rejecting (ver=%d) transaction %s at block height %d\n",
+            __func__, __LINE__, nVersion, GetHash().ToString(), nHeight);
+        return state.DoS(dosLevel, error("ContextualCheck(): tx are illegal at this height"),
+            CValidationState::Code::INVALID, "bad-txs-stopped");
+    }
+
```

Comment:
Minor thing here: I would suggest doing this check before the possibly expensive call to `CheckBlockAtHeight()`

Reply:
Ok, done